### PR TITLE
ENH: Add Azure Pipelines CI.

### PR DESCRIPTION
**Pull request by jhlegarreta**
Add Azure Pipelines CI, which supersedes Appveyor, Circle and Travis CI.

Update the `README` file badges accordingly.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
The target and image within the project template will surely need some revision. Thanks.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,86 +0,0 @@
...
-      - image: insighttoolkit/module-ci:latest
-    steps:
-      - checkout:
-          path: /ITKModuleTemplateCookieCutter
-      - run:
-          name: Fetch CTest driver script
-          command: |
-            curl -L https://raw.githubusercontent.com/InsightSoftwareConsortium/ITK/dashboard/itk_common.cmake -O
-      - run:
-          name: Evaluate template
```
[the remark this answers is not in the document]
Comment (jhlegarreta) [reply .circleci/config.yml]: I do not see the missing step in the added file. The `CTest` driver script is being fetched actually. Where do we need to add it ? Does this apply to all `azure-pipelines.yml` files? The [ITKSplitComponents azure-pipelines.yml](https://github.com/InsightSoftwareConsortium/ITKSplitComponents/blob/master/azure-pipelines.yml) does not include it  🤔 .

[the remark this answers is not in the document]
Comment (thewtex) [reply .circleci/config.yml]: Only the `ITKModuleTemplate` repository, we need to evaluate the template with the defaults,

```
python -m cookiecutter --no-input /ITKModuleTemplateCookieCutter
```

Compare *ITKModuleTemplate/.circleci/config.yml* with *ITKModuleTemplate/{{cookiecutter.project_name}}/.circleci/config.yml*.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,216 @@
+variables:
+  ITKGitTag: v4.13.1
```
[the remark this answers is not in the document]
Comment (jhlegarreta) [reply azure-pipelines.yml]: Used the ITK v5.0rc01 in c267ab4.

File: .circleci/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhlegarreta) [reply .circleci/config.yml]: Thanks for the explanation :+1: . Let's see if I did this correctly in c267ab4.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    displayName: 'Build ITK'
+
+  - script: |
+      curl -L https://raw.githubusercontent.com/InsightSoftwareConsortium/ITK/dashboard/itk_common.cmake -O
+    displayName: 'Fetch CTest driver script'
+
+  - script: |
+      python -m pip install cookiecutter
+      cd /
+      python -m cookiecutter --no-input /ITKModuleTemplateCookieCutter
```
Comment (thewtex) [inline azure-pipelines.yml]: Here we want `/ITKModuleTemplateCookieCutter` -> `--output-dir $(Agent.BuildDirectory)/ITKModuleTemplate $(Agent.SourcesDirectory)`

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+    displayName: 'Evaluate template'
+
+  - bash: |
+      cat > dashboard.cmake << EOF
+      set(CTEST_SITE "Azure.\$ENV{AGENT_MACHINENAME}")
+      file(TO_CMAKE_PATH "\$ENV{AGENT_BUILDDIRECTORY}" CTEST_DASHBOARD_ROOT)
+      file(TO_CMAKE_PATH "\$ENV{BUILD_SOURCESDIRECTORY}" CTEST_SOURCE_DIRECTORY)
```
Comment (thewtex) [inline azure-pipelines.yml]: Here we want `"\$ENV{BUILD_SOURCESDIRECTORY}"` -> `"\$ENV{AGENT_BUILDDIRECTORY}/ITKModuleTemplate"

Comment (jhlegarreta) [reply azure-pipelines.yml]: Done in a587ac1.

File: azure-pipelines.yml (hunk 3 above)
Comment (jhlegarreta) [reply azure-pipelines.yml]: Done in a587ac1.

**Comment by jhlegarreta** [comment]
@thewtex thanks :+1:

**Pull request merged**